### DISCUSSION TITLE
Are @app.route definitions synchronous?

### DISCUSSION THREAD

**Discussion by liuqijie6** [miguelgrinberg/Flask-SocketIO#1951] [Q&A]
```
from flask import Flask, render_template
from flask_socketio import SocketIO, emit
    
app = Flask(__name__)
app.config['SECRET_KEY'] = 'secret!'
socketio = SocketIO(app)

@app.route('/index')
def index():
    return "123"

@socketio.event
def my_event(message):
    emit('my response', {'data': 'got it!'})

if __name__ == '__main__':
    socketio.run(app)
```
I understand that @socketio.event is asynchronous, why @app.route is synchronous?
If I run with gunicron and set threads>1, will @app.route have any effect?

**Comment by miguelgrinberg** [comment]
I don't really understand the question. The concurrency of your Flask endpoints is controlled by your web server. If you run a web server in threaded mode and you set the thread limit to 1, then your application will only be able to process one request at a time.

In the context of your web server, Flask-SocketIO can be conceptually counted as one request. If your web server can only spawn one thread, then once a client connects over Socket.IO the web server will not be able to process any other clients (HTTP or Socket.IO) until that first Socket.IO connection is dropped.
